File: scripts/check-html.py
from html.parser import HTMLParser
from pathlib import Path

class Parser(HTMLParser):
    pass

for file_name in ['index.html', 'admin-dashboard.html', 'student-dashboard.html']:
    Parser().feed(Path(file_name).read_text(encoding='utf-8'))

print('HTML parse ok')

File: scripts/check-html.mjs
import fs from 'node:fs';

const files = ['index.html', 'admin-dashboard.html', 'student-dashboard.html'];

for (const file of files) {
  const content = fs.readFileSync(file, 'utf8');
  if (!content.includes('<!DOCTYPE html>') && !content.includes('<html')) {
    throw new Error(`Invalid HTML in ${file}`);
  }
}

console.log('HTML parse ok');
